### PR TITLE
fix(protocol-designer): address more 7.0 candidate a bugs

### PR DESCRIPTION
**Pull request by jerader**
closes RQA-1145, RQA-1163, RQA-1132, RQA-1161

# Overview

addresses more bugs for PD 7.0

# Test Plan

sandbox: https://sandbox.designer.opentrons.com/pd_address-more-7.0-bug-fixes/

make sure the bugs are addressed
1. when you click on the `settings` button, the app in the file side bar should not appear to be clickable ([RQA-1161](https://opentrons.atlassian.net/browse/RQA-1161))
2. for the OT-2, when you go through the file creator wizard and get to the modules page, when you select a module, a GEN should be selected by default (RQA-1145)
3. add a color, if you try to type in `000000` or `ffffff`, it should give you an error saying those colors are not allowed
4. the `whaleGrey` color isn't allowed to be selected as a color in the preselected colors

# Changelog

- add on to the color picker YUP validation to disallow certain colors
- add default modules for OT-2 protocols
- change the css for the app thing in setting, since it shouldn't behave like a button. Note: I commented out the css instead of removing it because we might add more options that would require it in the future

# Review requests

see test plan

# Risk assessment

low

[RQA-1161]: https://opentrons.atlassian.net/browse/RQA-1161?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by koji** [comment]
In terms of [RQA-1145](https://opentrons.atlassian.net/browse/RQA-1145),
https://sandbox.designer.opentrons.com/pd_address-more-7.0-bug-fixes/

When taking off focus, the wizard shows the warning and it says `required` but Review file details is clickable and actually I could finish the create wizard without selecting gen. Is that expected behavior?

https://github.com/Opentrons/opentrons/assets/474225/6e041751-0882-4e8b-b226-f024f625a0f4



[RQA-1145]: https://opentrons.atlassian.net/browse/RQA-1145?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by koji** [comment]
maybe disabled the color that uses form's background would be good? (not sure)

Screenshot 2023-08-02 at 6 30 21 PM

Screenshot 2023-08-02 at 6 34 43 PM

**Comment by koji** [comment]
whaleGrey is `#9395a0`, right? I could save it as a liquid color.

Screenshot 2023-08-02 at 6 40 01 PM

File: protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx [hunk 1]
```diff
@@ -43,7 +49,16 @@ export const liquidEditFormSchema: Yup.Schema<
       name: i18n.t('form.liquid_edit.name'),
     })
   ),
-  displayColor: Yup.string(),
+  displayColor: Yup.string().test(
+    'disallowed-color',
+    'Invalid display color',
+    value => {
+      if (value === null || value === undefined) {
```
Comment (koji) [inline protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx]:
```suggestion
      if (value == null) {
```

File: protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx [hunk 2]
```diff
@@ -84,81 +99,97 @@ export function LiquidEditForm(props: Props): JSX.Element {
...
+                      }}
+                    />
+                  </FormGroup>
+                </div>
+                <Flex
+                  justifyContent={JUSTIFY_END}
+                  color="#9e5e00"
+                  fontSize="0.625rem"
+                  fontWeight={TYPOGRAPHY.fontWeightSemiBold}
+                  marginTop={errors.name != null ? `-0.25rem` : `0rem`}
```
Comment (koji) [inline protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx]:
```suggestion
                  marginTop={errors.name != null ? '-0.25rem' : '0rem'}
```

If we could avoid using `-`, that would be great. But I guess this is really needed and the easiest way.

File: protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx [hunk 3]
```diff
@@ -84,81 +99,97 @@ export function LiquidEditForm(props: Props): JSX.Element {
...
+                <p className={styles.info_text}>
+                  {i18n.t('form.liquid_edit.serialize_explanation')}
+                </p>
+                <DeprecatedCheckboxField
+                  name="serialize"
+                  label={i18n.t('form.liquid_edit.serialize')}
+                  value={values.serialize}
+                  onChange={handleChange}
+                />
+              </section>
```
Comment (koji) [inline protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx:179]: Do we really need these spaces?

File: protocol-designer/src/components/SettingsPage/SettingsPage.css [hunk 4]
```diff
@@ -8,16 +8,15 @@
...
 
-.sidebar_item:hover:not(.disabled) {
+/* .sidebar_item:hover:not(.disabled) {
   outline: var(--bd-width-default) solid var(--c-highlight);
 }
 
 .sidebar_item.titled_list_selected .title_bar {
   background-color: var(--c-bg-selected);
-}
+} */
```
Comment (koji) [inline protocol-designer/src/components/SettingsPage/SettingsPage.css]: Don't need this any more?
If so, deleting these lines will be making this css file clean?

File: protocol-designer/src/components/SettingsPage/SettingsSidebar.tsx [hunk 5]
```diff
@@ -21,7 +21,7 @@ const SettingsSidebarComponent = (props: Props): JSX.Element => (
     <PDTitledList
       className={styles.sidebar_item}
       selected={props.currentPage === 'settings-app'}
-      onClick={props.makeNavigateToPage('settings-app')}
+      // onClick={props.makeNavigateToPage('settings-app')}
```
Comment (koji) [inline protocol-designer/src/components/SettingsPage/SettingsSidebar.tsx]: Need to keep this line?

**Comment by jerader** [comment]
> When taking off focus, the wizard shows the warning and it says `required` but Review file details is clickable and actually I could finish the create wizard without selecting gen. Is that expected behavior?
> 

@koji  for some reason my commit for this fix doesn't seem to be reflected? 👀 i'll look into it, thanks!

**Comment by jerader** [comment]
> whaleGrey is `#9395a0`, right? I could save it as a liquid color.
> 
> Screenshot 2023-08-02 at 6 40 01 PM

Ya good catch @koji , i need to add a bunch of colors to the "do not allow" list. Probably the 2 greys you mentioned and some other similar colors.

**Comment by koji** [comment]
> > whaleGrey is `#9395a0`, right? I could save it as a liquid color.
> > Screenshot 2023-08-02 at 6 40 01 PM
> 
> Ya good catch @koji , i need to add a bunch of colors to the "do not allow" list. Probably the 2 greys you mentioned and some other similar colors.

when I tried to set ffffff/000000, I could see the warning message, invalid color code.
However, when I set 000001, the app allowed me to use that color. Is that expected behavior?
Maybe checking the range of hex would be good?

like
```ts
function checkColor(hex: string): boolean {
  const cleanHex = hex.replace("#", "");
  // Convert the hex code to RGB values
  const red = parseInt(cleanHex.substr(0, 2), 16);
  const green = parseInt(cleanHex.substr(2, 2), 16);
  const blue = parseInt(cleanHex.substr(4, 2), 16);
  const luminance = (0.299 * red + 0.587 * green + 0.114 * blue) / 255;
  return (luminance < 0.1 || luminance > 0.9);
}

const hexCode1 = "#FFFFF1"; // White-ish color
const hexCode2 = "#000001"; // Black-ish color
const hexCode3 = "#FF0000"; // Not white-ish or black-ish color

console.log(hexCode1, checkColor(hexCode1));  true
console.log(hexCode2, checkColor(hexCode2)); true
console.log(hexCode3, checkColor(hexCode3)); false
```

File: protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx (hunk 3 above)
Comment (jerader) [reply protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx:179]: hmmm idk, prettier put them there 🤔 gonna leave it as is i think

File: protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx [hunk 6]
```diff
@@ -34,6 +38,18 @@ interface LiquidEditFormValues {
   serialize?: boolean
   [key: string]: unknown
 }
+
+function checkColor(hex: string): boolean {
+  const cleanHex = hex.replace('#', '')
+  const red = parseInt(cleanHex.slice(0, 2), 16)
+  const green = parseInt(cleanHex.slice(2, 4), 16)
+  const blue = parseInt(cleanHex.slice(4, 6), 16)
+  const luminance = (0.299 * red + 0.587 * green + 0.114 * blue) / 255
```
Comment (koji) [inline protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx:47]: #FFF0FF is invalid color so we will need to modify the values later.
But I guess most people don't use that type of the color.

File: protocol-designer/src/components/modals/CreateFileWizard/ModulesAndOtherTile.tsx [hunk 7]
```diff
@@ -112,7 +112,7 @@ export function ModulesAndOtherTile(props: WizardTileProps): JSX.Element {
       showHeaterShakerPipetteCollisions={showHeaterShakerPipetteCollisions}
     />
   )
-
+  console.log(values.modulesByType)
```
Comment (koji) [inline protocol-designer/src/components/modals/CreateFileWizard/ModulesAndOtherTile.tsx]: if this is for a debugging purpose, let's remove.

File: protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx [hunk 8]
```diff
@@ -84,81 +111,96 @@ export function LiquidEditForm(props: Props): JSX.Element {
...
+                      onChange={(color: ColorResult['hex']) => {
+                        setFieldValue('displayColor', color)
+                      }}
+                    />
+                  </FormGroup>
+                </div>
+                <Flex
+                  justifyContent={JUSTIFY_END}
+                  color="#9e5e00"
+                  fontSize="0.625rem"
```
Comment (koji) [inline protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx]: can use 

`TYPOGRAPHY.fontSizeCaption`, but I'm not sure that fontSizeCaption makes sense to PD.

**Review by koji — APPROVED** [review]
Tested all issues on the sandbox and this time everything worked as expected.
I left a couple of comments.

File: protocol-designer/src/components/modals/CreateFileWizard/ModulesAndOtherTile.tsx (hunk 7 above)
Comment (jerader) [reply protocol-designer/src/components/modals/CreateFileWizard/ModulesAndOtherTile.tsx]: yup definitely for debugging, thanks for catching it!

File: protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx (hunk 6 above)
Comment (jerader) [reply protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx:47]: I'll make a ticket for this and put it in the PD backlog. I think it definitely needs to be fleshed out more.

Comment (jerader) [reply protocol-designer/src/components/LiquidsPage/LiquidEditForm.tsx:47]: made a follow up ticket: https://opentrons.atlassian.net/browse/RAUT-620

**Pull request merged**